server/migrations.go

```diff
@@ -18,6 +28,12 @@ const (
 
 // runKVStoreMigrations checks the KV store version and runs necessary migrations
 func (p *Plugin) runKVStoreMigrations() error {
+	_, err := p.runKVStoreMigrationsWithResults()
+	return err
+}
+
+// runKVStoreMigrationsWithResults checks the KV store version and runs necessary migrations, returning detailed results
+func (p *Plugin) runKVStoreMigrationsWithResults() (*MigrationResult, error) {
```

Comment:
Could probably implement something along the lines the cloud provisioner is using in the future:  [store](https://github.com/mattermost/mattermost-cloud/blob/master/internal/store/migrate.go#L36) | [migrations](https://github.com/mattermost/mattermost-cloud/blob/master/internal/store/migrations.go)

And this is one thing we could add boilerplate to the pluginapi as well.

Reply:
Agreed.  This was a quick & dirty migration system needed only because I keep changing what is stored in KV.  These migrations aren't needed when this is released and will be removed soon.  If future versions need a migration we can use the pattern you linked to.